programs/protocol-contracts-solana/src/lib.rs

```diff
@@ -101,48 +97,41 @@ pub mod gateway {
         );
         system_program::transfer(cpi_context, amount)?;
         msg!(
-            "{:?} deposits {:?} lamports to PDA",
+            "{:?} deposits {:?} lamports to PDA; receiver {:?}",
             ctx.accounts.signer.key(),
-            amount
+            amount,
+            receiver,
         );
 
         Ok(())
     }
 
-    pub fn deposit_spl_token(
-        ctx: Context<DepositSplToken>,
+    pub fn deposit_and_call(
+        ctx: Context<Deposit>,
         amount: u64,
-        memo: Vec<u8>,
+        receiver: [u8; 20],
+        message: Vec<u8>,
     ) -> Result<()> {
-        require!(memo.len() >= 20, Errors::MemoLengthTooShort);
-        require!(memo.len() <= 512, Errors::MemoLengthExceeded);
-        let token = &ctx.accounts.token_program;
-        let from = &ctx.accounts.from;
+        require!(message.len() <= 512, Errors::MemoLengthExceeded);
```

Comment:
Can we increase this limit?

Reply:
to 4096?

Reply:
adjusted to 1024 there is not reason to go beyond 1024
as that's the limit of transaction size on solana.